web/settings.py

```diff
@@ -81,7 +81,7 @@
 
 REST_FRAMEWORK = {
     "DEFAULT_AUTHENTICATION_CLASSES": [
-        "rest_framework.authentication.SessionAuthentication",
+        "rest_framework.authentication.BasicAuthentication",
```

Comment:
Why was this changed? I don't believe this is necessary, and basic authentication is generally only appropriate for testing purposes. Furthermore, django-rest-registration is designed around session/token based authentication.

Reply:
@alinaah You did not address this comment.

Reply:
This still has not been addressed.

Reply:
I can't do anything on this, otherwise Apis won't work. 
It is better to use token authentication. It is not a common practis of uning session authentication with apis. Session authentication do not work with apis, instead we should use token authentication.

Reply:
> It is better to use token authentication.

No, in this case it is not. Token authentication makes sense for non-browser clients, but our client is the browser. Furthermore, basic authentication, which you have configured, is _not_ token authentication.

However, if you specifically meant a JSON Web Token, then it's a different story. But it'd require more work to set up, and regardless, I am doubtful of the benefits it provides in our case over session auth.

> It is not a common practis of uning session authentication with apis.

It is common and is the best choice when the client of the API is a browser.

> I can't do anything on this, otherwise Apis won't work.
> Session authentication do not work with apis

Yes, they do work. I have tested it and even demonstrated it in class during a presentation. I also left several comments on your previous PR with examples of how to authenticate using sessions by getting a CSRF token from a cookie and passing it as a header.

Reply:
In order to do Session authentication i need to add csrf tokens. If it was  normal django template in frontend  then i can use {{csrf_token}}. But now the case is different we are using angular. {{csrf_token}} in <form> tag is not gonna work. You need to pass csrf token from backend when a user logged in. But i could n't find any.

Reply:
Yes all that is taken care of. I left a comment on your old PR which showed how to do this https://github.com/Comp-490-SeniorProject/site/pull/43#issuecomment-1045227851

Please note that you may need to adjust the code, as that was just an example in pure JS. Angular may have its own utilities for managing cookies, and of course its own way to make http requests (which you're already using in this PR).

Reply:
You still need to change this back to `SessionAuthentication`. When I was testing your PR on my machine, I changed it back (together with removing the `0.0.0.0` base URL) and the register request still worked perfectly fine.